toml/src/main/kotlin/org/rust/toml/crates/local/CratesLocalIndexServiceImpl.kt

```diff
@@ -215,37 +220,57 @@ class CratesLocalIndexServiceImpl
         }
 
         val revTree = RevWalk(repository).parseCommit(ObjectId.fromString(currentHeadHash)).tree
+        val mapper = JsonMapper()
+            .registerKotlinModule()
+            .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)
 
         TreeWalk(repository).use { treeWalk ->
             treeWalk.addTree(revTree)
             treeWalk.filter = filter
             treeWalk.isRecursive = true
             treeWalk.isPostOrderTraversal = false
 
+            val objectIds = mutableListOf<Pair<ObjectId, String>>()
             while (treeWalk.next()) {
                 if (treeWalk.isSubtree) continue
                 if (treeWalk.nameString == "config.json") continue
 
                 val objectId = treeWalk.getObjectId(0)
-                val loader = repository.open(objectId)
-                val versions = mutableListOf<CargoRegistryCrateVersion>()
-                val fileReader = BufferedReader(InputStreamReader(loader.openStream()))
-
-                fileReader.forEachLine { line ->
-                    if (line.isBlank()) return@forEachLine
-
+                objectIds.add(objectId to treeWalk.nameString)
+            }
+            val pool = Executors.newWorkStealingPool(2)
+            val future = supplyAsync(pool) {
+                objectIds
+                    .parallelStream()
+                    .map { (objectId, name) ->
+                        val loader = repository.open(objectId)
+                        val versions = mutableListOf<CargoRegistryCrateVersion>()
+                        val fileReader = BufferedReader(InputStreamReader(loader.openStream()))
```

Comment:
```suggestion
                        val fileReader = BufferedReader(InputStreamReader(loader.openStream(), Charsets.UTF_8))
```
I accidentally noticed that we forget to specify `Charsets.UTF_8` charset here(